roles/mysql/tasks/main.yml

```diff
@@ -1,19 +1,44 @@
 ---
 # tasks file for mysql
 #
+- name: MySQL | remove mysql-client
+  ansible.builtin.apt:
+    name: mysql-client
+    state: absent
+    autoremove: true
+
+- name: MySQL | ensure keyrings directory exist
+  ansible.builtin.file:
+    path: /etc/apt/keyrings
+    state: directory
+    mode: "0755"
+
 - name: MySQL | Install MariaDB key
-  ansible.builtin.apt_key:
+  ansible.builtin.get_url:
```

Comment:
Possibly relevant here - see the first example in the [docs](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_key_module.html#ansible-collections-ansible-builtin-apt-key-module) on `apt_key`.